pkgs/applications/misc/activitywatch/default.nix

```diff
@@ -0,0 +1,49 @@
+{ config, lib, stdenv, fetchzip, qt6, wrapGAppsNoGuiHook, glib-networking, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "ActivityWatch";
+  version = "0.12.1";
+  meta = with lib; {
+    description = "ActivityWatch is an app that automatically tracks how you spend time on your devices.";
+    longDescription = ''
+      ActivityWatch is an app that automatically tracks how you spend time on your devices.
+      It is open source, privacy-first, cross-platform, and a great alternative to services like RescueTime, ManicTime, and WakaTime.
+      It can be used to keep track of your productivity, time spent on different projects, bad screen habits, or just to understand how you spend your time.
+    '';
+    homepage = "https://github.com/ActivityWatch/activitywatch";
+    license = licenses.mpl20;
+    platforms = platforms.linux;
+  };
+
+  src = fetchzip {
+    url = "https://github.com/ActivityWatch/activitywatch/releases/download/v${version}/activitywatch-v${version}-linux-x86_64.zip";
+    hash = "sha256-VdnAKcLQBWBVZqRyD1xoYuAfvTeCtcFJp3/slkQ6ynk=";
+  };
+
+  buildInputs = [ qt6.qtbase glib-networking ];
+  nativeBuildInputs = [ autoPatchelfHook wrapGAppsNoGuiHook ];
```

Comment:
Why not compile from source?

Reply:
activitywatch is composed of a lot of small services, the derivation proposed on the related issue is 380 lines long

Reply:
which is preferable if someone already has done the work

Reply:
I'll try, what do you recommend for node packages? buildNpmPackage?